lib/swoosh/adapters/mailgun.ex

```diff
@@ -79,19 +79,23 @@ defmodule Swoosh.Adapters.Mailgun do
   end
   defp prepare_custom_vars(body, _email), do: body
 
+  defp prepare_attachments(body, %{attachments: []}), do: body
+  defp prepare_attachments(body, %{attachments: attachments}) do
+    Map.put(body, :attachment, Enum.map(attachments, &File.read!(&1.path)))
+  end
 
-  defp prepare_from(body, %Email{from: from}), do: Map.put(body, :from, prepare_recipient(from))
```

Comment:
Sorry to be annoying @princemaple but if we make that change I'd rather make it across all adapters and in a separate PR for clarity. Does that make sense?